From ec36df9d1fbb6dcbf0d1e79245ffe213049ecd5a Mon Sep 17 00:00:00 2001
From: Dongxiao Xu <dongxiao.xu@intel.com>
Date: Wed, 3 Aug 2011 14:33:07 +0800
Subject: [PATCH] connman-dbus: fix security type mismatch

Latest connman change the Security type to be array{String}, fix
connman-gnome accordingly.

Signed-off-by: Dongxiao Xu <dongxiao.xu@intel.com>
---
 common/connman-dbus.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

Upstream-Status: Pending

diff --git a/common/connman-dbus.c b/common/connman-dbus.c
index 0f4e1db..0ac8ed5 100644
--- a/common/connman-dbus.c
+++ b/common/connman-dbus.c
@@ -384,7 +384,8 @@ static void service_changed(DBusGProxy *proxy, const char *property,
 		gtk_tree_store_set(store, &iter,
 					CONNMAN_COLUMN_FAVORITE, favorite, -1);
 	} else if (g_str_equal(property, "Security") == TRUE) {
-		const char *security = value ? g_value_get_string(value) : NULL;
+		const char **array = value ? g_value_get_boxed(value) : NULL;
+		const char *security = g_strjoinv(" ", array);
 		gtk_tree_store_set(store, &iter,
 					CONNMAN_COLUMN_SECURITY, security,
 					-1);
@@ -487,7 +488,7 @@ static void service_properties(DBusGProxy *proxy, GHashTable *hash,
 	strength = value ? g_value_get_uchar(value) : 0;
 
 	value = g_hash_table_lookup(hash, "Security");
-	security = value ? g_value_get_string(value) : NULL;
+	security = value ? g_strjoinv(" ", g_value_get_boxed(value)) : NULL;
 
 	value = g_hash_table_lookup(hash, "PassPhrase");
 	passphrase = value ? g_value_get_string(value) : NULL;
-- 
1.7.1

